server/work-queue/queue.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var nr = require('node-resque');
+var Q = require('q');
+
+var config = require('../config');
+var log = require('../logger')('server.work-queue');
+var jobs = require('./jobs');
+
+var queueReady = Q.defer();
+var workersReady = Q.defer();
+
+var connectionDetails = {
+  host: config.get('db.redis.host'),
+  password: config.get('db.redis.password'),
+  port: config.get('db.redis.port'),
+  database: config.get('db.redis.database')
+};
+
+var queueOpts = { connection: connectionDetails };
+var queue = new nr.queue(queueOpts, jobs, function onQueueReady() {
```

Comment:
ESLint says: "A constructor name should not start with a lowercase letter".
Probably not worth caring about since it seems to object to the "queue" part of it, so we'd have to refactor to something absurd like:
```js
var Queue = nr.queue;
var MultiWorker = nr.multiworker;

var queue = new Queue(queueOpts, jobs, function {...});
```